src/unaudited/ECDSAStakeRegistry.sol

```diff
@@ -0,0 +1,385 @@
+    /// @return uint256 - The weight of the operator at the given block.
+    function getOperatorWeightAtBlock(
+        address _operator,
+        uint32 _blockNumber
+    ) external view returns (uint256) {
+        return _operatorWeightHistory[_operator].getAtBlock(_blockNumber);
+    }
+
+    /// @notice Retrieves the total weight at a specific block number.
+    /// @param _blockNumber The block number to get the total weight for the quorum
+    /// @return uint256 - The total weight at the given block.
+    function getLastCheckpointTotalWeightAtBlock(
+        uint32 _blockNumber
+    ) external view returns (uint256) {
+        return _totalWeightHistory.getAtBlock(_blockNumber);
+    }
+
+    function operatorRegistered(address _operator) external view returns (bool) {
+        return _operatorRegistered[_operator];
+    }
+
+    /// @notice Calculates the current weight of an operator based on their delegated stake in the strategies considered in the quorum
+    /// @param _operator The address of the operator.
+    /// @return uint256 - The current weight of the operator; returns 0 if below the threshold.
+    function getOperatorWeight(address _operator) public view returns (uint256) {
+        StrategyParams[] memory strategies = _quorum.strategies;
+        uint256 weight;
+        uint256 sharesAmount;
+        for (uint256 i; i < strategies.length; i++) {
+            sharesAmount = DELEGATION_MANAGER.operatorShares(_operator, strategies[i].strategy);
+            weight += sharesAmount * strategies[i].multiplier;
+        }
+        weight = weight / BPS;
+        return weight >= minimumWeight ? weight : 0;
+    }
+
+    /// @notice Initializes state for the StakeRegistry
+    /// @param _serviceManager The AVS' ServiceManager contract's address
+    function __ECDSAStakeRegistry_init(
+        address _serviceManager,
+        uint256 _thresholdWeightBps,
+        Quorum memory _quorum
+    ) internal onlyInitializing {
+        serviceManager = _serviceManager;
+        _updateStakeThreshold(_thresholdWeightBps);
+        _updateQuorumConfig(_quorum);
+        __Ownable_init();
+        __Pausable_init();
+    }
+
+    function _updateStakeThreshold(uint256 _thresholdWeightBps) internal {
+        _thresholdWeightBpsHistory.push(_thresholdWeightBps);
+        emit ThresholdWeightUpdated(_thresholdWeightBps);
+    }
+
+    /// @dev Internal function to set the quorum configuration
+    /// @param _newQuorum The new quorum configuration to set
+    function _updateQuorumConfig(Quorum memory _newQuorum) internal {
+        if (!_isValidQuorum(_newQuorum)) revert InvalidQuorum();
+        Quorum memory oldQuorum = _quorum;
+        delete _quorum;
+        for (uint256 i; i < _newQuorum.strategies.length; i++) {
+            _quorum.strategies.push(_newQuorum.strategies[i]);
+        }
+        emit QuorumUpdated(oldQuorum, _newQuorum);
+    }
```

Comment:
seems like this is only updating the `strategies` and not the `weights` ?

I took some inspiration from your struct definition + wrote a similar function here https://github.com/Layr-Labs/eigenlayer-middleware/blob/970349466df340dc87c79fb3029f37c6b464a646/src/experimental/SimpleLinearWeightQuorum.sol#L59-L86

Reply:
Need to look into the origin of the naming here, but the `strategies` array is an array of `strategyParams`.  The multiplier in the `strategyParmas` is effectively the weight

Reply:
ahhhhh my bad! OK will resolve thread.